config/nas_mirror.sh

```
@@ -0,0 +1,84 @@
+SNAP_COUNT=$(echo "$SRC_SNAPS" | wc -l)
+if [ "$SNAP_COUNT" -eq 0 ]; then
+    log "ERROR: Source snapshot list empty ($SOURCE). Aborting."
+    exit 1
+fi
+
+# ---- Delete obsolete snapshots ----
+DELETED=0
+for snap in "${DEST_SNAPS[@]}"; do
+    if [[ -z "${SRC_SET[$snap]:-}" ]]; then
+        log "Deleting obsolete snapshot: $snap"
+        rm -rf "$DEST/$snap"
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)